### ISSUE TITLE
4.2 Instruction fine-tuning on Q&A dataset

### ISSUE THREAD

**Issue by ns-1456** [ns-1456/kisan-saathi#14]
### Objective
Fine-tune domain-adapted model on Q&A instruction dataset.

### Requirements
- Load domain-adapted model from previous phase
- Fine-tune on Q&A instruction dataset
- Learning rate: 1e-4 (lower than domain adaptation)
- Epochs: 3-5, batch size: 4
- Monitor instruction-following capability

### Acceptance Criteria
- [ ] Instruction tuning completed
- [ ] Model performance evaluated
- [ ] Quality improvements verified
- [ ] Final model saved
- [ ] Evaluation report created

### Estimated Time
2-3 days

### Dependencies
Domain adaptation training

### Sprint
Sprint 4: LoRA Fine-tuning
